src/Bicep.Core/Registry/ModuleDispatcher.cs

```diff
@@ -30,29 +31,50 @@ public ModuleDispatcher(IModuleRegistryProvider registryProvider)
 
         public ImmutableArray<string> AvailableSchemes { get; }
 
-        public ModuleReference? TryGetModuleReference(string reference, out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder)
+        public ModuleReference? TryGetModuleReference(string reference, RootConfiguration configuration, out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder)
         {
-            var parts = reference.Split(':', 2, System.StringSplitOptions.None);
+            var parts = reference.Split(':', 2, StringSplitOptions.None);
             switch (parts.Length)
             {
                 case 1:
                     // local path reference
                     if (registries.TryGetValue(ModuleReferenceSchemes.Local, out var localRegistry))
                     {
-                        return localRegistry.TryParseModuleReference(parts[0], out failureBuilder);
+                        return localRegistry.TryParseModuleReference(parts[0], configuration, out failureBuilder);
                     }
 
                     failureBuilder = x => x.UnknownModuleReferenceScheme(ModuleReferenceSchemes.Local, this.AvailableSchemes);
                     return null;
 
                 case 2:
-                    var scheme = parts[0];
+                    string scheme = parts[0];
+                    string? aliasName = null;
+
+                    if (parts[0].Contains("/"))
+                    {
+                        // The sheme contains an alias.
+                        var schemeParts = parts[0].Split('/', 2, StringSplitOptions.None);
+                        scheme = schemeParts[0];
+                        aliasName = schemeParts[1];
+                    }
 
                     if (!string.IsNullOrEmpty(scheme) && registries.TryGetValue(scheme, out var registry))
                     {
                         // the scheme is recognized
                         var rawValue = parts[1];
-                        return registry.TryParseModuleReference(rawValue, out failureBuilder);
+
+                        if (aliasName is not null)
+                        {
+                            if (configuration.ModuleAliases.TryGetModuleAlias(scheme, aliasName, out failureBuilder) is not { } alias)
+                            {
+                                return null;
+                            }
+
+                            rawValue = $"{alias}/{rawValue}";
+                        }
+
+
+                        return registry.TryParseModuleReference(rawValue, configuration, out failureBuilder);
```

Comment:
>rawValue [](http://example.com/codeflow?start=64&length=8)

The errors that we return from `TryParseModuleReference()` include the `rawValue` in the message. 

When we have an alias we are creating a new raw value that isn't present in the Bicep file and you can't Ctrl+F for it because the parts of it are split between config and the bicep file.

It it going to confuse people? (They will still have the text span that points them to the location.)

Reply:
Updated the error messages to use the format of
```
The OCI artifact / Template Spec reference "{referenceValue}“ after resolving alias “{aliasName}” is not valid...
```
when there exists an alias in the raw reference.